test/Microsoft.DotNet.Docker.Tests/DockerHelper.cs

```diff
@@ -0,0 +1,105 @@
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Docker.Tests
+{
+    public class DockerHelper
+    {
+        private string DockerOS => GetDockerOS();
+        public string ContainerWorkDir => IsLinuxContainerModeEnabled ? "/sandbox" : "c:\\sandbox";
+        public bool IsLinuxContainerModeEnabled => string.Equals(DockerOS, "linux", StringComparison.OrdinalIgnoreCase);
+        private ITestOutputHelper Output { get; set; }
+
+        public DockerHelper(ITestOutputHelper output)
+        {
+            Output = output;
+        }
+
+        public void Build(string dockerfilePath, string fromImage, string tag, string buildArgs)
+        {
+            string dockerfileContents = File.ReadAllText(dockerfilePath);
+            dockerfileContents = dockerfileContents.Replace("$base_image", fromImage);
+            dockerfilePath = dockerfilePath + ".temp";
+            File.WriteAllText(dockerfilePath, dockerfileContents);
+
+            try
+            {
+                Execute($"build -t {tag} {buildArgs} -f {dockerfilePath} .");
+            }
+            finally
+            {
+                File.Delete(dockerfilePath);
+            }
+        }
+
+        public void DeleteImage(string name)
+        {
+            if (ResourceExists("image", name))
+            {
+                Execute($"image rm -f {name}");
+            }
+        }
+
+        public void DeleteVolume(string name)
+        {
+            if (ResourceExists("volume", name))
+            {
+                Execute($"volume rm -f {name}");
+            }
+        }
+
+        private void Execute(string args)
+        {
+            Output.WriteLine($"Executing : docker {args}");
+            ProcessStartInfo info = new ProcessStartInfo("docker", args);
+            info.RedirectStandardOutput = true;
+            info.RedirectStandardError = true;
+            Process process = Process.Start(info);
+            process.WaitForExit();
+            Output.WriteLine(process.StandardOutput.ReadToEnd());
+
+            if (process.ExitCode != 0)
+            {
+                string stdErr = process.StandardError.ReadToEnd();
+                string msg = $"Failed to execute {info.FileName} {info.Arguments}{Environment.NewLine}{stdErr}";
+                throw new InvalidOperationException(msg);
+            }
+        }
+
+        private static string GetDockerOS()
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo("docker", "version -f \"{{ .Server.Os }}\"");
+            startInfo.RedirectStandardOutput = true;
+            Process process = Process.Start(startInfo);
+            process.WaitForExit();
+            return process.StandardOutput.ReadToEnd().Trim();
+        }
+
+        public string GetContainerWorkPath(string relativePath)
+        {
+            string separator = IsLinuxContainerModeEnabled ? "/" : "\\";
```

Comment:
Instead of `"\\"` why not `$"{Path.DirectorySeparatorChar}"`

Reply:
I would prefer to not make an assumption that the separator within the container the not the same as the host OS.

Reply:
Valid point. Thanks!